File: src/CodeChef/oct18b/mindsum.py
def digSum(n):
    ret = 0
    while n > 0:
        ret += n % 10
        n = n // 10
    return ret


def calc(n, d):
    count = 1000
    ans, steps = 11, float('inf')
    visited = {}
    current = [n]
    level = 0
    while count > 0:
        count -= 1
        next = []
        for x in current:
            if x < ans:
                ans = x
                steps = level
            elif x == ans and level < steps:
                steps = level
            if x + d not in visited:
                visited[x + d] = 1
                next.append(x + d)
            digS = digSum(x)
            if digS not in visited:
                visited[digS] = 1
                next.append(digS)
        level += 1
        current = next[:]
    return str(ans) + " " + str(steps)


if __name__ == '__main__':
    for _ in range(int(input().strip())):
        n, d = [int(__) for __ in input().strip().split()]
        print(calc(n, d))

File: src/CodeChef/oct18b/Mindsum.java
//package CodeChef.oct18b;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Scanner;

/**
 * Created by bk on 07-10-2018.
 */
class Mindsum {
    public static void main(String[] __) {
        Scanner sc = new Scanner(System.in);
        int tc = sc.nextInt();
        while (tc-- > 0) {
            long n = sc.nextLong(), d = sc.nextLong();
            System.out.println(calc(n, d));
        }
    }

    private static String calc(long n, long d) {
        int count = 100000;
        long ans = 11, steps = Long.MAX_VALUE;
        HashMap<Long, Integer> visited = new HashMap<>();
        List<Long> current = List.of(n);
        int level = 0;
        while (count-- > 0) {
            List<Long> next = new ArrayList<>();
            for (long x : current) {
                if (x < ans) {
                    ans = x;
                    steps = level;
                } else if (x == ans && level < steps) steps = level;
                if (!visited.containsKey(x + d)) {
                    visited.put(x + d, 1);
                    next.add(x + d);
                }
                long digS = digSum(x);
                if (!visited.containsKey(digS)) {
                    visited.put(digS, 1);
                    next.add(digS);
                }
            }
            level += 1;
            current = new ArrayList<>(next);
        }
        return ans + " " + steps;
    }

    private static long digSum(long n) {
        long ret = 0;
        while (n > 0) {
            ret += n % 10;
            n = n / 10;
        }
        return ret;
    }

}
